pkg/virt-operator/kubevirt_test.go

```diff
@@ -1604,10 +1680,43 @@ var _ = Describe("KubeVirt Operator", func() {
 			kubecontroller.SetLatestApiVersionAnnotation(kv)
 			addKubeVirt(kv)
 			addInstallStrategy(defaultConfig)
-			addAll(defaultConfig)
+			addAll(defaultConfig, kv)
```

Comment:
We have a test case here that verifies that nothing will change if a resource's generation does not change "should do nothing if KubeVirt object is deployed". That's great, we have that part covered.

I'd like to see a unit test that exercises the logic specifically around strictly reconciling a resource when the generation ID doesn't match what's on the kubevirt cr. 

That way we'll have covered both sides... the do nothing side (already covered now) and the should do something side